solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -372,6 +372,11 @@ private StatefulCollectionWatch compute(
 
   private static class StatefulCollectionWatch extends CollectionWatch<DocCollectionWatcher> {
     private DocCollection currentState;
+    private volatile StateWatcher associatedWatcher;
```

Comment:
Can we add some description for this field, there are `DocCollectionWatcher` and `StateWatcher` here. Perhaps it's good to mention that this is the underlying ZK level watcher that we tracking. And we need this to stop re-installing old watches .